lib/Driver/Job.cpp

```diff
@@ -352,7 +353,11 @@ void Job::printCommandLineAndEnvironment(raw_ostream &Stream,
 void Job::printCommandLine(raw_ostream &os, StringRef Terminator) const {
   escapeAndPrintString(os, Executable);
   os << ' ';
-  printArguments(os, Arguments);
+  if (hasResponseFile()) {
+    printArguments(os, ResponseFileArg);
+  } else {
+    printArguments(os, Arguments);
+  }
```

Comment:
If the above is right, then I think it should be:
`printArguments(os, hasResponseFile() ? ResponseFileArg : Arguments);` rather than the if.
Why? Because the code is only doing *one* `printArguments` operation, and it is exactly the same, except for that second argument.